internal/nakamabenchmark/testsuite_test.go

```diff
@@ -0,0 +1,391 @@
+package nakamabenchmark
```

Comment:
What are we trying to do with the `nakamabenchmark` package? It seems like `/nakamabenchmark/testsuite_test.go` and `/nakamabenchmark/util.go` both have a bunch of commented out code?

Reply:
This is a test file for the future. It's copied over from the e2e tests. The reason why it's still here because while you're running the benchmark one can run the e2e tests as well. 

In the e2e tests one can run things like external http clients. This allows for realistic benchmarking under realistic loads. The code to do this will go here in the future and it exists in the form of a "testing" file. 

Right now it doesn't do anything, but it's here for the future. I commented out the code because it can be used to create test code as a reference.

Reply:
@namrapatel if you want me to just delete the files and comments, I can do that. I don't feel particularly strongly about this one.

Reply:
I see, I'm personally OK with leaving this commented code here for use in the near future, insofar as there is a ticket open for that work. I say we let Jeremy and/or Tyler chime in on this and make a final decision, I'm in favour of leaving it in and opening a ticket

Reply:
There is no ticket. Because we literally don't know what we need to benchmark in the future. 

I only know that we WILL benchmark something in the future. Commenting the code was a practical action here, I realize it's "ugly" but in the future if you do need to write some clients, the file and references are right here. 

Whatever, I can delete it if you feel your right and I'm wrong.

Reply:
Nah, I'm cool with small amounts of commented code as long as we think it's actually good, I see what you mean re: no currently opened ticket. Unless someone else takes issue with this in the next day I think you can resolve this one

Reply:
I don't agree with having commented out files in our code base (this goes for the [game/query/location.go](https://github.com/Argus-Labs/world-engine/pull/540/files#diff-c62d41ff1949849882121655f821c850fd2324c9fcab3f2e590732fc5c6761d3) file above). 

We can make a couple of triage tickets in Linear referencing the source of code code, titled something like "Consider Query benchmarks" and in the body "Use the query code at internal/e2e/tester/game/query for an example".

Reply:
Are you only referring to query? That's more easily removable. @jerargus

Reply:
@jerargus, removed. This one is less less critical. I feel more strongly about the test files, we can just wait for Tylers vote on those.

Reply:
Also guys, none of this is nearly as bad as cancelling the entire events feature imo and leaving the dead code in there lol.

Reply:
I think both game/query/location.go and this entire internal/nakamabenchmark directory should be removed.

Reply:
@jerargus How about this. I add general code that will load test benchmark in there. A single test and that will make the code relevant for the e2e benchmark.

Reply:
lmk if you guys are all cool with that and I'll do it.

Reply:
I think you should leave this out and add a triage ticket to Linear to benchmark queries and a separate triage ticket to benchmark Nakama CQL and Events stuff.

Reply:
You're saying do this in a separate PR? @jerargus. I would agree with that. The problem is I'll be blocked on this until way later. I think a quick test here to make the load more realistic in the profile works. 

We can vote on it. @technicallyty what do you think?

Reply:
agree with jeremy, would prefer to just throw the commented out code into a triage ticket. the difference between this and the events situation is, we know what to do and have a ticket for addressing the events. there is a plan, and people are assigned and know what to do when the server stuff is completed. here, the dead code is just here for a rainy day.

i vote to toss it into triage as jeremy suggested above

Reply:
@namrapatel @jerargus @technicallyty 

I talked to tyler in person, He seemed convinced, it seems that from that conversation he and most people aren't fully grasping what I'm doing with the profiling PR here. It's not possible to fully understand what this is if you weren't following my conversations with Scott on slack which is fine, I made the mistake of assuming people were but there's too much going on here. 

I'll explain it more in depth in stand up tomorrow. And I'll show you the results of it, it's actually very visual and pretty cool.

Reply:
As discussed I added the http client to the profiling. You can see the load it adds to the profile here:

![Screenshot 2024-01-24 at 12 30 47 PM](https://github.com/Argus-Labs/world-engine/assets/1806857/b2cfbeb9-a53e-46d4-9c1c-68abccf354a8)

Just want to reiterate. The nature of the flame graph makes it pointless to test things by themselves. So if I test just redis, or just parsing, or just the http server I get no context. The flamegraph only gives you benchmarking info about things Relative to OTHER things. So you actually have to benchmark more than one thing to even get this style of profiling to work.